sympy/core/mul.py

```diff
@@ -487,12 +492,15 @@ def _gather(c_powers):
         num_rat = []
         for e, b in comb_e.items():
             b = cls(*b)
-            if e.q == 1:
-                coeff *= Pow(b, e)
+            c = Pow(b, e)
+            if e.q == 1 and c.is_Number:
+                coeff *= c
                 continue
             if e.p > e.q:
                 e_i, ep = divmod(e.p, e.q)
-                coeff *= Pow(b, e_i)
+                c = Pow(b, e_i)
+                if c.is_Number:
+                    coeff *= Pow(b, e_i)
```

Comment:
No point computing this twice
```suggestion
                    coeff *= c
```